yrs/src/transact.rs

```diff
@@ -233,3 +233,44 @@ pub enum TransactionAcqError {
     #[error("All references to a parent document containing this structure has been dropped.")]
     DocumentDropped,
 }
+
+#[cfg(test)]
+mod test {
+    use crate::{Doc, GetString, Text, Transact};
+    use rand::random;
+    use std::sync::{Arc, Barrier};
+    use std::time::{Duration, Instant};
+
+    #[test]
+    fn multi_thread_transact_mut() {
+        let doc = Doc::new();
+        let txt = doc.get_or_insert_text("text");
+
+        const N: usize = 3;
+        let barrier = Arc::new(Barrier::new(N + 1));
+
+        let start = Instant::now();
+        for i in 0..N {
+            let d = doc.clone();
+            let t = txt.clone();
+            let b = barrier.clone();
+            std::thread::spawn(move || {
+                // let mut txn = d.try_transact_mut().unwrap(); // this will hang forever
```

Comment:
From [this comment](https://github.com/y-crdt/y-crdt/pull/481#issue-2472045080):
> 1. `Transact::transact` and `Transact::transact_mut` methods now won't panic when transaction couldn't be acquired. Instead they will block a thread. You can always simulate old behaviour by using `doc.try_transact().unwrap()`.

I understood that `d.try_transact_mut().unwrap()` will panic (for all other threads after the first one acquired the transaction), not that it will hang forever?

Reply:
In this case it will hang forever, since panic inside of the thread will prevent execution of `barrier.wait()`, which means that we won't be able to pass over the barrier on the main test thread.